R/functions.R

```diff
@@ -0,0 +1,345 @@
+        factor(),
+      .by = taakomschrijving
+    ) |>
+    pivot_longer(
+      cols = karen_1:datamanager_2,
+      names_to = c(".value", "task_type"),
+      names_pattern = "(.+)_([12])$"
+    ) |>
+    pivot_longer(
+      karen:datamanager,
+      names_to = "person",
+      values_to = "nr_days",
+      values_drop_na = TRUE
+    ) |>
+    mutate(
+      task_type = ifelse(task_type == "1", "uitvoering", "review") |> fct(),
+      nr_days = ifelse(
+        continuous,
+        nr_days,
+        nr_days / nr_months
+      ),
+      across(where(is.character), fct)
+    ) |>
+    select(-c(first_day_currentmonth:end, nr_months))
+}
+
+
+#' @keywords internal
+summarize_planning_long <- function(x,
+                                    restrict_to_selected = TRUE,
+                                    priorities = 1:10,
+                                    max_year = max_y(),
+                                    tempres = c("y_month", "y"),
+                                    include_continuous = TRUE) {
+  tempres <- if (!is.null(tempres)) (match.arg(tempres))
+  x |>
+    mutate(y = year(date)) |>
+    filter(
+      if (restrict_to_selected) doen_we else TRUE,
+      prioriteit %in% priorities,
+      y <= max_year,
+      include_continuous | !continuous
+    ) |>
+    summarize(
+      days = sum(nr_days) |> round(2),
+      .by = c(person, all_of(tempres))
+    ) |>
+    arrange(person, {{ tempres }})
+}
+
+#' Summarize planning (days occupied) by person and by month or year
+#'
+#' @param planning_long Long format of planning data.
+#' @param restrict_to_selected Logical.
+#' Should only the tasks where `doen_we` is `TRUE` be kept?
+#' If `FALSE`, then all tasks are kept.
+#' @param priorities Numeric vector of priorities, used to filter the
+#' 'prioriteit' column in x.
+#' @param max_year Number; the maximum allowed year from planning_long.
+#' @param tempres String. Temporal resolution of the result.
+#' @param include_continuous Logical.
+#' Should continuous tasks be included?
+#' Defaults to `TRUE`; value `FALSE` can be useful in manual checks or
+#' debugging.
+summarize_planning <- function(planning_long,
+                               restrict_to_selected = TRUE,
+                               priorities = 1:10,
+                               max_year = max_y(),
+                               tempres = c("y_month", "y"),
+                               include_continuous = TRUE) {
+  tempres <- if (!is.null(tempres)) (match.arg(tempres))
+  summarize_planning_long(
+    x = planning_long,
+    restrict_to_selected = restrict_to_selected,
+    priorities = priorities,
+    max_year = max_year,
+    tempres = tempres,
+    include_continuous = include_continuous
+  ) |>
+    pivot_wider(
+      names_from = person,
+      values_from = days,
+      names_sort = TRUE
+    )
+}
+
+#' Update the plansum_xxx sheets in the planning googlesheet
+#'
+#' @inheritParams get_planning_long
+#' @inheritParams summarize_planning
```

Comment:
I think this function has a different signature (resulting in wrong docs with `inheritParams`)